Fixes DeadObjectException

nitaliano opened the pull request:
* ~~Force TextureView on Android to prevent ANR/DeadObjectException since the Android SDK is expecting `onDestroy` to be called in the parents Activities `onDestroy` with a GLSurfaceView~~

~~One thing to note is there is a ticket open in `gl-native` to figure out a way to get the `GLSurfaceView` and `onDestroy` to work without needing to be called in the parent activity, so we are still keeping that code around because we expect it to work in future releases~~

See https://github.com/mapbox/react-native-mapbox-gl/pull/1033#issuecomment-366581754

Commit:
Add onStart/Resume/Pause/Stop/Destroy hooks

cpetzel commented on android/rctmgl/src/main/java/com/mapbox/rctmgl/components/mapview/RCTMGLMapView.java:
```diff
@@ -249,11 +250,23 @@ public AbstractMapFeature getFeatureAt(int i) {
+            public void run() {
+                mapView.onDestroy();
+            }
+        });
```
Because 'this' is a view, you could just post on its message queue. You can also just call the onDestroy method directly. 

```
post(new Runnable() {
            @Override
            public void run() {
                onDestroy();
            }
        });
```

nitaliano replied:
o yeah duh, I was thinking I was in the manager here I'll call it directly 👍

Commit:
[Android] Adds custom shadow node for MapView called MapShadowNode

This shadow node will inform us when we are about to dispose of our NativeMap
and we will destroy the NativeMapView when we destroy our ShadowNode to avoid
the ANR while calling NativeMapView.onDestroy
